### PR TITLE
#625: RqMultipartTest.java:91-98: This test is using RqGreedy...

### PR DESCRIPTION
**Pull request by 2686747**
#625 fixed: refactored RqMultipart.Fake, removed RqGreedy from tests

### CODE REVIEW

**Comment by davvd** [comment]
@2686747 I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by davvd** [comment]
@dmzaytsev review this one, please

File: src/main/java/org/takes/rq/RqMultipart.java [hunk 1]
```diff
@@ -569,18 +522,59 @@ private static StringBuilder fakeBody(final Request... parts)
...
+                );
             }
+
+            /**
+             * Fake body creator.
+             * @param parts Fake request body parts
+             * @return StringBuilder of given dispositions
+             * @throws IOException If fails
+             */
+            @SuppressWarnings("PMD.InsufficientStringBufferDeclaration")
```
Comment (dmzaytsev) [inline src/main/java/org/takes/rq/RqMultipart.java]: @2686747 could you get rid the suppression?

File: src/main/java/org/takes/rq/RqMultipart.java [hunk 2]
```diff
@@ -569,18 +522,59 @@ private static StringBuilder fakeBody(final Request... parts)
...
+            private static StringBuilder fakeBody(final Request... parts)
+                throws IOException {
+                final StringBuilder builder = new StringBuilder();
+                for (final Request part : parts) {
+                    builder.append(
+                        String.format("--%s", FakeMultipartRequest.BOUNDARY)
+                        )
+                        .append(RqMultipart.CRLF)
+                        // @checkstyle MultipleStringLiteralsCheck (1 line)
+                        .append("Content-Disposition: ")
```
Comment (dmzaytsev) [inline src/main/java/org/takes/rq/RqMultipart.java]: @2686747 actually only one string constant `Content-Disposition: ` in the class
I think we don't need this supression

File: src/main/java/org/takes/rq/RqMultipart.java [hunk 3]
```diff
@@ -569,18 +522,59 @@ private static StringBuilder fakeBody(final Request... parts)
...
+                        .append(
+                            new RqHeaders.Smart(
+                                new RqHeaders.Base(part)
+                            // @checkstyle MultipleStringLiteralsCheck (1 line)
+                            ).single("Content-Disposition")
```
Comment (dmzaytsev) [inline src/main/java/org/takes/rq/RqMultipart.java]: @2686747 let's make a constant on class level for `Content-Disposition` header name
and get rid the supression

**Comment by dmzaytsev** [comment]
@2686747 please see my comments

**Comment by 2686747** [comment]
@dmzaytsev check my fixes, please

**Comment by dmzaytsev** [comment]
@2686747 thanks
what about pmd suppression?

**Comment by 2686747** [comment]
> what about pmd suppression?

@dmzaytsev i'm not sure about it: there is no performance issue, a length in StringBuilder's constructor seems pointless, but if you have other arguments... i have nothing.

**Comment by dmzaytsev** [comment]
@2686747 there is an aim to get rid suppressions if possible.
in this case it is very simple, just add a capacity to the constructor.
anyway feel free to create a ticket in Qulice project to remove this PMD rule

**Comment by dmzaytsev** [comment]
@2686747 thanks

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@yegor256 ping

**Comment by dalifreire** [comment]
@davvd I think we have a misunderstand here... you assigned the task #625 to me but we have this PR waiting for @yegor256 to approve the merge. Let me know if you want me to go ahead...

File: src/main/java/org/takes/rq/RqMultipart.java [hunk 4]
```diff
@@ -64,7 +64,14 @@
  */
 @SuppressWarnings("PMD.TooManyMethods")
 public interface RqMultipart extends Request {
-
+    /**
+     * Carriage return constant.
+     */
+    String CRLF = "\r\n";
```
Comment (yegor256) [inline src/main/java/org/takes/rq/RqMultipart.java]: @2686747 this is not acceptable, see http://www.yegor256.com/2015/07/06/public-static-literals.html

**Comment by yegor256** [comment]
@dmzaytsev can't merge this, the code is not good, see my comment above. I thought you're aware of that "public static literals" problem

**Comment by dmzaytsev** [comment]
@yegor256 I am aware of that problem, but the interface does not allow to declare private constants. Java 6 doesn't allow default implementations in interfaces as well.
So I have chosen the public constant instead of a checkstyle suppression.
In my opinion one public constant looks better than multiply checkstyle suppressions.

**Comment by yegor256** [comment]
@dmzaytsev it's possible to add one `@checkstyle MultipleStringLiteralsCheck (500 lines)` to the javadoc of the entire class

**Comment by dmzaytsev** [comment]
@yegor256 actually it is against the idea of that warning I guess

anyway @2686747 please remove `CRLF` constant and put the checkstyle suppression  as yegor256 suggested above

**Comment by dmzaytsev** [comment]
@2686747 please merge with master as well

**Comment by 2686747** [comment]
@dmzaytsev sorry, is not my task now.

**Comment by dmzaytsev** [comment]
@yegor256 would it possible assign @2686747 to #625 back?
or what do we have to do with this PR?

**Comment by dmzaytsev** [comment]
@2686747 could you close the PR please?  
we can't continue here because  #625 resolved already by another developer

**Comment by 2686747** [comment]
@dmzaytsev sure, no problems.

**Comment by davvd** [comment]
@ypshenychka review this ticket please, for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by ypshenychka** [comment]
@2686747 
According to our [QA Rules](http://at.teamed.io/qa.html):
>Pull request title explains the problem it is fixing.

Title is simply copied from puzzle and doesn't provide any information. Please provide meaningful title.

**Comment by ypshenychka** [comment]
@2686747 Please respond to my remark.

**Comment by 2686747** [comment]
@ypshenychka Sorry, it's not my task. And i don't work with the team by the way.

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@ypshenychka thanks a lot, next time everybody should try to make it better

**Comment by davvd** [comment]
@dmzaytsev 10 mins was added to the account of @ypshenychka (for QA review), in transaction 86387762; Thanks so much! Your account was topped up for **29 mins** (transaction ID is `86387791`, the task took 759 hours and 49 mins); you're getting extra minutes here (c=14); +29 added to your rating, current score is: [+1226](http://www.netbout.com/b/35166?open=rating)

**Pull request closed without merging**